From 6e4a35b4de81556e768dc7a5b1a2c3a8983ef155 Mon Sep 17 00:00:00 2001
From: Gordon Sim <gsim@apache.org>
Date: Tue, 13 Jan 2015 10:25:08 +0000
Subject: [PATCH 08/12] QPID-6310: handle case where content bearing method is
 not in fact a message-transfer, which is the only one currently supported by
 qpidd

git-svn-id: https://svn.apache.org/repos/asf/qpid/trunk@1651320 13f79535-47bb-0310-9956-ffa450edef68
---
 qpid/cpp/src/qpid/broker/MessageBuilder.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/qpid/cpp/src/qpid/broker/MessageBuilder.cpp b/qpid/cpp/src/qpid/broker/MessageBuilder.cpp
index 7cb9951..f5e9332 100644
--- a/qpid/cpp/src/qpid/broker/MessageBuilder.cpp
+++ b/qpid/cpp/src/qpid/broker/MessageBuilder.cpp
@@ -45,6 +45,9 @@ void MessageBuilder::handle(AMQFrame& frame)
     switch(state) {
     case METHOD:
         checkType(METHOD_BODY, type);
+        if (!frame.getMethod()->isA<qpid::framing::MessageTransferBody>())
+            throw NotImplementedException(QPID_MSG("Unexpected method: " << *(frame.getMethod())));
+
         exchange = frame.castBody<qpid::framing::MessageTransferBody>()->getDestination();
         state = HEADER;
         break;
-- 
2.1.0

